File: app/src/main/java/ch/bailu/aat/map/layer/grid/UTMCenterCoordinatesLayer.java
package ch.bailu.aat.map.layer.grid;

import org.mapsforge.core.model.LatLong;

import ch.bailu.aat.coordinates.MeterCoordinates;
import ch.bailu.aat.coordinates.UTMCoordinates;

public class UTMCenterCoordinatesLayer extends CenterCoordinatesLayer{
    @Override
    public MeterCoordinates getCoordinates(LatLong p) {
        return new UTMCoordinates(p);
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/map/layer/grid/UTMCenterCoordinatesLayer.kt
package ch.bailu.aat_lib.map.layer.grid

import ch.bailu.aat_lib.coordinates.MeterCoordinates
import ch.bailu.aat_lib.coordinates.UTMCoordinates
import ch.bailu.aat_lib.preferences.StorageInterface
import ch.bailu.aat_lib.service.ServicesInterface
import org.mapsforge.core.model.LatLong

class UTMCenterCoordinatesLayer(services: ServicesInterface, storage: StorageInterface) :
    CenterCoordinatesLayer(services, storage) {
    override fun getCoordinates(p: LatLong): MeterCoordinates {
        return UTMCoordinates(p)
    }
}
